### PR TITLE
tests: no indentation on json.dumps fails

### PR DESCRIPTION
**Pull request by greut**
This is confusing.

**Idea** using the real dict to store the stuff and hack around to get the _special_ behaviors.

### CODE REVIEW

File: dojson/contrib/marc21/utils.py [hunk 1]
```diff
@@ -30,48 +30,30 @@ def __new__(cls, values=None):
...
-                        keys.append(key)
-                    if occurences[key] > 1:
-                        value = values[key][counter[key]]
-                    else:
-                        value = values[key]
-                    data.append((key, value))
-                    counter[key] += 1
+                new = dict.copy(values)
             else:
+                tmp = {}
```
Comment (jirikuncar) [inline dojson/contrib/marc21/utils.py]: `defaultdict(list)`

File: dojson/contrib/marc21/utils.py [hunk 2]
```diff
@@ -30,48 +30,30 @@ def __new__(cls, values=None):
...
+                ordering = []
+
                 for key, value in values:
-                    if key not in keys:
-                        keys.append(key)
-                    data.append((key, value))
+                    if key not in tmp:
```
Comment (jirikuncar) [inline dojson/contrib/marc21/utils.py]: after https://github.com/inveniosoftware/dojson/pull/56/files#r47129235 it should not be necessary

File: tests/test_contrib_marc21.py [hunk 3]
```diff
@@ -29,26 +29,38 @@ def test_groupable_ordered_dict_is_immutable(god):
...
+    keys = god.keys()
+    keys.sort()
+    assert ['a', 'b', 'c'] == keys
+    assert ['a', 'b', 'c', 'a', 'b'] == god.keys(repeated=True)
 
 
 def test_groupable_ordered_dict_items(god):
     """Test that a GroupableOrderedDict has items like a dict, but more."""
-    assert (('a', ('dojson', 4)), ('b', (2, 5)), ('c', 'invenio')) == god.items()
+    assert ('a', ('dojson', 4)) in god.items()
```
Comment (jirikuncar) [inline tests/test_contrib_marc21.py]: - `god_items = god_items()` so we don't call it 4 times

**Comment by greut** [comment]
@jirikuncar I've reworked it to use an `OrderedDict`. It gives ordering for free on some methods (which was lost after the previous refactoring).

**Pull request merged**